clientconn.go

```diff
@@ -225,21 +225,16 @@ func Dial(target string, opts ...DialOption) (*ClientConn, error) {
 // cancellation and expiration of ctx will be noop. Users should call ClientConn.Close
 // to terminate all the pending operations after this function returns.
 // This is the EXPERIMENTAL API.
-func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *ClientConn, err error) {
+func DialContext(ctx context.Context, target string, opts ...DialOption) (_ *ClientConn, err error) {
 	cc := &ClientConn{
 		target: target,
 		conns:  make(map[Address]*addrConn),
 	}
 	cc.ctx, cc.cancel = context.WithCancel(context.Background())
+
 	defer func() {
-		select {
-		case <-ctx.Done():
-			if conn != nil {
-				conn.Close()
-			}
-			conn = nil
-			err = ctx.Err()
-		default:
+		if err != nil {
```

Comment:
This might make the returned error inconsistent. I propose:

```
if err != nil {
  cc.Close()
}
select {
  case <-ctx.Done():
    err = ctx.Err()
  default:
  }
```